### DISCUSSION TITLE
Use a custom HttpClientHandler to do fake/custom DNS resolution

### DISCUSSION THREAD

**Discussion by maurodx** [dotnet/yarp#1425] [Q&A] [answered]
I'm trying to use reverse-proxy to route requests to different server that respond on the same domain name.

Two different server: 
a.a.a.a that reply to request for https://mysite.com
b.b.b.b that reply to request for https://mysite.com

only a.a.a.a can be resolved via DNS lookup.

How can I route https://localhost/first/*** to a.a.a.a and https://localhost/second/*** to b.b.b.b if I need to proxy out the request always with the same destination host url?

With an HttpClientHandler I can put my DNS resolution logic looking for some custom variables.

**Comment by samsp-msft** [comment, marked as answer]
The URL sent in the SNI and Host header fields can be configured independently of the name/IP of the destination host. By default they are set to be the same, but can be configured with transforms:

* If you want the original host to be forwarded: https://microsoft.github.io/reverse-proxy/articles/transforms.html#requestheaderoriginalhost
* If you want to manually configure the host: https://microsoft.github.io/reverse-proxy/articles/transforms.html#requestheader and set a value for Host.

**Reply by maurodx** [reply]
It works!
Adding .WithTransformRequestHeader(headerName: "Host", value: "mysite.com", append: false) to RouteConfig and using ip address as cluster destination address.
